File: api/Models/Exercise.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models;

[Table("Exercise")]
public class Exercise
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    public string CreatedById { get; set; }
    public AppUser CreatedBy { get; set; }

    public int PictureId { get; set; }
    public FileInfo Picture { get; set; }
    
    public int VideoId { get; set; }
    public FileInfo Video { get; set; }

    public bool IsApproved { get; set; } = false;

    // Foreign key for ExerciseLevel
    public int ExerciseLevelId { get; set; }
    // Navigation property
    public ExerciseLevel ExerciseLevel { get; set; }

    // Many-to-Many with muscle group
    public ICollection<MuscleGroup> MuscleGroups { get; set; } = new HashSet<MuscleGroup>();
}

File: frontend/src/models/Exercise.ts
export type ExerciseLevel = {
    id: number,
    name: string
}

export type MuscleGroup = {
    id: number,
    name: string
}

export type Exercise = {
    id: number,
    title: string,
    description: string,
    pictureId: number,
    videoId: number,
    exerciseLevel: ExerciseLevel,
    muscleGroups: MuscleGroup[],
    isApproved: boolean
}

export type ExerciseUpload = {
    title: string;
    description: string;
    picture: File;
    video: File;
    exerciseLevelId: number;
    muscleGroupIds: number[];
}